src/sentry/taskworker/client.py

```diff
@@ -163,7 +182,13 @@ def update_task(
         )
         try:
             with metrics.timer("taskworker.update_task.rpc"):
-                response = self._stub.SetTaskStatus(request)
+                if task_id not in self._task_id_to_stub_idx:
+                    metrics.incr("taskworker.client.task_id_not_in_client")
+                    return None
+                stub_idx = self._task_id_to_stub_idx[task_id]
+                response = self._stubs.pop(self._task_id_to_stub_idx[task_id]).SetTaskStatus(
+                    request
+                )
```

Comment:
Wouldn't `self._stubs.pop()` remove grpc-channels preventing them from being used again?

```suggestion
                stub_idx = self._task_id_to_stub_idx.pop(task_id, None)
                if stub_idx is None:
                    metrics.incr("taskworker.client.task_id_not_in_client")
                    return None
                response = self._stubs[stub_idx].SetTaskStatus(
                    request
                )
```

I think we want to pop from the `_task_id_to_stub_idx` map and not `_stubs`.

Reply:
Ugh yeah you're totally right, can't believe I missed that. My brain is already on holiday mode, my bad